src/node_alive/autostarter.py

```diff
@@ -0,0 +1,119 @@
+# ROS
+import os
+import roslaunch
+import rospkg
+import rospy
+from std_msgs.msg import String
+
+# TU/e Robotics
+from node_alive.srv import AutoStarterCommand, AutoStarterCommandRequest, AutoStarterCommandResponse
+
+class AutoStarter(object):
+    """
+    Launches a launch file and waits for a command to either shutdown or to restart.
+    It is also possible to start other launch files that are stored on the parameter server, like the demo's.
+    """
+    launch_files_list = []
+
+    def __init__(self):
+        """
+        Loads the launch files that can be started from the parameter server.
+        Creates a service over which the command will be received, a publisher that publishes the current launch file.
+        Set some variables.
+        """
+        self.launch_files_list = rospy.get_param('~launch_files_list')
+        self._command_service = rospy.Service('auto_starter_command', AutoStarterCommand, self._srv_callback)
+        self._status_pub = rospy.Publisher('current_launch_file', String, queue_size=1, latch=True)
+
+        self.launch = None
+        self._pending_req = None
+        self._pending_resp = None
+
+    def update(self):
+        """
+        If there is a request of the service that was pending, it will be executed in the main thread when sleep is over
+        After executing the request, the pending status will be reset.
+        :return:
+        """
+        if self._pending_req is not None:
+            self._pending_resp = self._handle_auto_starter_command(self._pending_req)
+            self._pending_req = None
+
+    def _srv_callback(self, req):
+        """
+        Recieves the request from the service and sets it to pending. As long as there is no response, the function
+        will sleep till it gets one. After that the response will reset.
+        :param req: the command request that the service receives from the client contains two data types.
+        :return:
+        """
+        self._pending_resp = None
+        self._pending_req = req
+        rate = rospy.Rate(10.0)
+        while self._pending_resp is None and not rospy.is_shutdown():
```

Comment:
Maybe this should set pending resp back to None. I.e.,
```
response = self._pending_resp
self._pending_resp = None
return response
```

Reply:
Updated in a1ef39d19ac8416def3517506f5d571fcb9e1645